ocs_ci/ocs/platform_nodes.py

```diff
@@ -818,6 +824,271 @@ def _prepare_node(self, node_id):
             except Exception:
                 logger.error("Failed to create RHEL node")
                 raise
+        elif self.node_type == 'RHCOS':
+            conf = self._prepare_rhcos_node_conf()
+            conf['node_id'] = node_id
+            try:
+                self.aws_instance_obj = self._prepare_upi_rhcos_node(conf)
+            except Exception:
+                logger.error("Failed to create RHCOS node")
+                raise
+            self.approve_pending_csr()
+
+    def _prepare_rhcos_node_conf(self):
+        """
+        Merge default RHCOS node configuration for rhcos node
+        along with the user provided config
+
+        Returns:
+            conf (dict): A dictionary of merged user and default values
+
+        """
+        conf = self.read_default_config(constants.RHCOS_WORKER_CONF)
+        default_conf = conf.get('ENV_DATA')
+        merge_dict(default_conf, self.node_conf)
+        logger.info(f"Config after merge is {default_conf}")
+        return default_conf
+
+    def _prepare_upi_rhcos_node(self, conf):
+        """
+        Handle RHCOS worker instance creation using cloudformation template,
+        Create RHCOS instance with ami same as master
+
+        Args:
+            conf (dict): configuration for node
+
+        Returns:
+           awsinstance object
+
+        """
+        worker_template_path = self.get_rhcos_worker_template()
+        self.bucket_name = constants.AWS_S3_UPI_BUCKET
+        self.template_obj_key = 'workertemplate'
+        self.add_cert_to_template(worker_template_path)
+        self.aws.upload_file_to_s3_bucket(
+            self.bucket_name, self.template_obj_key, worker_template_path
+        )
+        s3_url = self.aws.get_s3_bucket_object_url(
+            self.bucket_name, self.template_obj_key
+        )
+        self.create_stack(conf, s3_url)
+        instance_id = self.aws.get_stack_instance_id(
+            self.stack_name, constants.AWS_WORKER_LOGICAL_RESOURCE_ID
+        )
+
+        delete_file(worker_template_path)
+        return self.aws.get_ec2_instance(instance_id)
+
+    def create_stack(self, conf, s3_url):
```

Comment:
I think that this method should be added to ocs_ci/utility/aws.py instead as it is interacting directly with boto3 and kind of low level implementation

Reply:
Moved to utility/aws.py